docs/guides/http2.md

```diff
@@ -0,0 +1,110 @@
+---
+title: HTTP/2
+description: Boost web performance and secure your communications by utilizing HTTP/2 with Tailcall.
+---
+
+[HTTP/2](https://www.cloudflare.com/en-in/learning/performance/http2-vs-http1.1/) represents a significant advancement in web protocols, offering enhancements such as multiplexing, header compression, and streamlined connection handling, all designed to improve web performance significantly.
+
+Tailcall enables support for HTTP/2 for both server (ingress) and client (egress) operations, facilitating the protocol's adoption for both incoming and outgoing server requests. For egress, there is no special setting that's required, Tailcall will automatically upgrade the connection to HTTP/2 whenever possible. For ingress there are a few setting that one needs to provide.
+
+## SSL
+
+For Tailcall to serve GraphQL over HTTP/2 we need to first enable SSL for which we need to generate a certificate and a key. To generate the required certificates (cert.pem and key.pem) [OpenSSL](https://www.openssl.org/source/) is a widely used option. Here are the steps to get started with SSL:
+
+1. **Install OpenSSL**: Download and install OpenSSL from its official [website](https://www.openssl.org/source/) if it's not already installed on your system.
+
+2. **Generate Private Key**
+
+   ```bash
+   openssl genrsa -out key.pem 2048
+   ```
+
+   This creates a 2048-bit RSA private key, storing it in a file named `key.pem`.
+
+3. **Generate Certificate Signing Request (CSR)**
+
+   ```bash
+   openssl req -new -key key.pem -out csr.pem
+   ```
+
+   You will be prompted to provide information for the certificate, such as the Common Name (CN), organization details, and locality. This information is embedded into the CSR, saved in a file named `csr.pem`. This file can be used to request a certificate from a Certificate Authority (CA) or generate a self-signed certificate.
+
+   ![http2-csr.png](../../static/images/docs/http2-csr.png)
+
+4. **Generate Self-Signed Certificate**
+
+   ```bash
+   openssl x509 -req -days 365 -in csr.pem -signkey key.pem -out cert.pem
+   ```
+
+   This generates a self-signed certificate valid for 365 days using the CSR from step 3 and the private key from step 2. The validity period can be adjusted by changing the number of days (-days). A "Signature ok" prompt confirms the successful creation.
+
+   ![http/2-cert.png](../../static/images/docs/http2-cert.png)
```

Comment:
can you please update the image such that the BG is transparent? Shadows are fine but without the grey background.